plugin/src/main/java/com/denizenscript/denizen/utilities/Utilities.java

```diff
@@ -582,29 +583,41 @@ public static ElementTag enumLikeToLegacyElement(Object val) {
         throw new UnsupportedOperationException("Cannot get legacy name element, value isn't an enum: " + val);
     }
 
-    // TODO: need proper input backsupport, see https://discord.com/channels/315163488085475337/1011496047811506227/1301272242386370580
-    public static <T> T elementToEnumlike(ElementTag element, Class<T> type) {
-        if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21) && Keyed.class.isAssignableFrom(type)) {
-            return (T) Bukkit.getRegistry((Class<? extends Keyed>) type).get(parseNamespacedKey(element.asString()));
+    @SuppressWarnings("unchecked")
+    public static <T extends Keyed> T elementToEnumlike(ElementTag element, Class<T> type, TagContext context) {
+        if (NMSHandler.getVersion().isAtMost(NMSVersion.v1_20)) {
+            return (T) element.asEnum((Class<? extends Enum<?>>) type);
+        }
+        Registry<T> registry = Bukkit.getRegistry(type);
+        T value = registry.get(parseNamespacedKey(element.asString()));
+        if (!Settings.cache_legacySpigotNamesSupport || value != null) {
+            return value;
+        }
+        String updatedName = NMSHandler.instance.updateLegacyName(type, element.asString());
+        if (CoreUtilities.equalsIgnoreCase(element.asString(), updatedName)) {
+            return null;
+        }
+        if (context != null) {
+            BukkitImplDeprecations.oldSpigotNames.warn(context);

```

Comment:
is it really worth having to pass a context around everywhere just for a deprecation notice?